File: mifosng-android/src/main/java/com/mifos/mifosxdroid/offline/syncclientpayloads/SyncClientPayloadsMvpView.java
package com.mifos.mifosxdroid.offline.syncclientpayloads;

import com.mifos.mifosxdroid.base.MvpView;
import com.mifos.objects.client.ClientPayload;

import java.util.List;

/**
 * Created by Rajan Maurya on 08/07/16.
 */
public interface SyncClientPayloadsMvpView extends MvpView {

    void showPayloads(List<ClientPayload> clientPayload);

    void showError(int stringId);

    void showSyncResponse();

    void showClientSyncFailed(String errorMessage);

    void showOfflineModeDialog();

    void showClientPayloadUpdated(ClientPayload clientPayload);

    void showPayloadDeletedAndUpdatePayloads(List<ClientPayload> clientPayloads);
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/offline/syncclientpayloads/SyncClientPayloadsMvpView.kt
package com.mifos.mifosxdroid.offline.syncclientpayloads

import com.mifos.mifosxdroid.base.MvpView
import com.mifos.objects.client.ClientPayload

/**
 * Created by Rajan Maurya on 08/07/16.
 */
interface SyncClientPayloadsMvpView : MvpView {
    fun showPayloads(clientPayload: List<ClientPayload>)
    fun showError(stringId: Int)
    fun showSyncResponse()
    fun showClientSyncFailed(errorMessage: String)
    fun showOfflineModeDialog()
    fun showClientPayloadUpdated(clientPayload: ClientPayload)
    fun showPayloadDeletedAndUpdatePayloads(clientPayloads: List<ClientPayload>)
}